### PR TITLE
Implementation of dynamic ProtoLoader

### PR DESCRIPTION
**Pull request by vmax**

### CODE REVIEW

File: tools/proto_tool/ProtoLoader.java [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+    final Path tempDirectory = Files.createTempDirectory("protoLoader");
+
+    ArrayList<String> args = new ArrayList<>(protos.size() + 3);
+    args.add("/usr/bin/env");
+    args.add("bash");
+    args.add(Paths.get("./tools/protoc.sh").toAbsolutePath().toString());
+
+    for (Path proto : protos) {
+      Path resutingFile =
```
Comment (oferb) [inline tools/proto_tool/ProtoLoader.java]: `resutingFile->resultingFile`

File: tools/proto_tool/ProtoLoader.java [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+      String updatedProto =
+          Files.readAllLines(proto)
+              .stream()
+              .map(
+                  s -> {
+                    if (s.contains("import")) {
+                      if (s.lastIndexOf('/') != -1) {
+                        // proto import contains directories
+                        s =
+                            String.format(
```
Comment (oferb) [inline tools/proto_tool/ProtoLoader.java]: Can you describe what you're doing to the imports?
Are you removing the directory part of the path?
If so, please explain why.

File: tools/proto_tool/ProtoLoader.java [hunk 3]
```diff
@@ -50,9 +50,17 @@ public static void toProtobin(
...
+      Path resultingFile =
           Paths.get(tempDirectory.toAbsolutePath().toString(), proto.getFileName().toString());
 
+      /*
+       * Proto files may import other proto files
+       * In order to support this, on copying them to temporary directory,
+       * we *rewire* the imports to have flat structure
+       * Example: `import common/files/something.proto` becomes `import something.proto`
+       * Current limitation of this approach is not supporting files with same name
+       * by silently overwriting them.
```
Comment (oferb) [inline tools/proto_tool/ProtoLoader.java]: Can you easily throw an error when we have a collision?

Comment (vmax) [reply tools/proto_tool/ProtoLoader.java]: @oferb fixed in https://github.com/google/startup-os/pull/574/commits/1ad957b11fed5f79e36d6508d709d1830fdc9bd8

File: tools/proto_tool/ProtoLoader.java (hunk 1 above)
Comment (vmax) [reply tools/proto_tool/ProtoLoader.java]: @oferb fixed in https://github.com/google/startup-os/pull/574/commits/0f4f6d6d49274a7362ee1d0270ec3c60ce3a7d39

File: tools/proto_tool/ProtoLoader.java (hunk 2 above)
Comment (vmax) [reply tools/proto_tool/ProtoLoader.java]: @oferb fixed in https://github.com/google/startup-os/pull/574/commits/0f4f6d6d49274a7362ee1d0270ec3c60ce3a7d39

**Pull request merged**